tests/backends/storage/test_swift.py

```diff
@@ -0,0 +1,291 @@
+
+    def mock_successful_stat(*args, **kwargs):  # pylint:disable=unused-argument
+        return {"success": True}
+
+    monkeypatch.setattr(SwiftService, "stat", mock_successful_stat)
+    try:
+        swift()
+    except Exception:  # pylint:disable=broad-except
+        pytest.fail("SwiftStorage should not raise exception on successful connection")
+
+
+@pytest.mark.parametrize("pages_count", [1, 2])
+def test_swift_list_should_yield_archive_names(
+    pages_count, swift, monkeypatch, fs
+):  # pylint:disable=invalid-name
+    """Given SwiftService.list method successfully connects to the Swift storage
+    The SwiftStorage list method should yield the archives
+    """
+
+    listing = [
+        {"name": "2020-04-29.gz"},
+        {"name": "2020-04-30.gz"},
+        {"name": "2020-05-01.gz"},
+    ]
+    history = [
+        {"id": "2020-04-29.gz", "backend": "swift", "command": "fetch"},
+        {"id": "2020-04-30.gz", "backend": "swift", "command": "fetch"},
+    ]
+
+    def mock_list_with_pages(*args, **kwargs):  # pylint:disable=unused-argument
+        return [{"success": True, "listing": listing}] * pages_count
+
+    def mock_successful_stat(*args, **kwargs):  # pylint:disable=unused-argument
+        return {"success": True}
+
+    monkeypatch.setattr(SwiftService, "list", mock_list_with_pages)
+    monkeypatch.setattr(SwiftService, "stat", mock_successful_stat)
+    fs.create_file(HISTORY_FILE, contents=json.dumps(history))
+    swift = swift()
+    assert list(swift.list()) == [x["name"] for x in listing] * pages_count
+    assert list(swift.list(new=True)) == ["2020-05-01.gz"] * pages_count
+    assert list(swift.list(details=True)) == listing * pages_count
+
+
+@pytest.mark.parametrize("pages_count", [1, 2])
+def test_swift_list_with_failed_connection_should_log_the_error(
+    pages_count, swift, monkeypatch, fs, caplog
+):  # pylint:disable=invalid-name
+    """Given SwiftService.list method fails to retrieve the list of archives
+    The SwiftStorage list method should log the error and raise a BackenException
+    """
+
+    def mock_list_with_pages(*args, **kwargs):  # pylint:disable=unused-argument
+        return [
+            {
+                "success": False,
+                "container": "ralph_logs_container",
+                "error": "Container not found",
+            }
+        ] * pages_count
+
+    def mock_successful_stat(*args, **kwargs):  # pylint:disable=unused-argument
+        return {"success": True}
+
+    monkeypatch.setattr(SwiftService, "list", mock_list_with_pages)
+    monkeypatch.setattr(SwiftService, "stat", mock_successful_stat)
+    fs.create_file(HISTORY_FILE, contents=json.dumps([]))
+    caplog.set_level(logging.ERROR)
+    swift = swift()
+    msg = "Failed to list container ralph_logs_container: Container not found"
+    with pytest.raises(BackendException, match=msg):
+        assert list(swift.list()) == []
```

Comment:
You don't need to assert something as once the exception has been raised you're out of the context.

```suggestion
        swift.list()
```

Reply:
Good catch) thanks)